Formula/r/readsb.rb

```diff
@@ -0,0 +1,65 @@
+class Readsb < Formula
+  desc "ADS-B decoder swiss knife"
+  homepage "https://github.com/wiedehopf/readsb"
+  url "https://github.com/wiedehopf/readsb/archive/refs/tags/v3.14.1666.tar.gz"
+  sha256 "395063fdc4ba70af14b5a782c230b94dd2bc7c19a6922c935ea8465ddaddc717"
+  license "GPL-3.0-or-later"
+
+  depends_on "pkgconf" => :build
+  depends_on "librtlsdr"
+  depends_on "zstd"
+
+  uses_from_macos "ncurses"
+
+  on_macos do
+    depends_on "libusb"
+  end
+
+  def install
+    ENV["RTLSDR"] = "yes"
+    system "make"
+    bin.install "readsb"
+  end
+
+  service do
+    run [opt_bin/"readsb", "--no-interactive", "--net"]
+  end
+
+  test do
+    require "base64"
+
+    enc = <<~EOS
+      goB/en99g4B9gH6Eg35/fICVhJl/gXx9f4F+f36Af4B/f3+AgH9+f39/gH+Af4CBf4F/f3
+      9/f3+AgIGAgH9/f4F/gX9/gH+Bf4B/f3+Af4F+f359gX5/f36BgYB9fIeGm46Ig3p+mIGYfX19f
+      oGFf4B/gIGAf3+AgoR8dHVle3iBhH98fn6AgIB+gIB/gIF/f4F+gYCAf3+AfoB/fX6CgICAen6K
+      i5WQhIF6foOBgYB9g454lW6CfH2Cg36AgYCBgX6Af4CAf4B/gICAf39/gICAf39/gICAf35/f4B
+      /f35+fn6Af4B/f4B/gIB+f35+gX+BfoF/f39/fYF/fIGDgYF/aHttfISAgIB9gYGBfn6AgIF8fn
+      6DlIaXgYF+fn+Df3+AgH+Cf4CAgH+Af4CAf4B+f4CAgH9+f39/foGCgH9xbG1wf4KDgHl/hIGCf
+      mSIc4mYfJB9jHSUb4N+e36CgYCCfmp/a3p1cmh3dYGHf39+foCAfX9+gIF9cIddm2ybhIV/e3qZ
+      gqyBkYiJjZSHipOJlYp9fIN+pn+ncol3eoGJd5lgjVt+eoCEe2V3aHt5Z2RfY3R8bntjen6BgYB
+      hiFmZc46FeXuMd5x/iX+Siq2PmH17hoWUk42GkoGVgpF7k3yOd49wjnSGbIdrhHB8Z3xwfHBsWG
+      Vlf4N/gGR4bXlwg2eIdoJpkFyfd417jHWdg4h+eYSal6qNkIyFk42PhZaAkYOOeplyhnl3fpBvm
+      FeEYn6Ff3x4ZHlydnNjYmNwfIN/gGN9UYVli4CBfH1mk2OmeZmDfnuFhqaQpIaIjoyWkYF+f32d
+      h6t7knV7got4lGmEd4hsi1CAaXx5dGR4dYOGcXFncn2CgoF7gH+Cf4F+f4CAf4GAgYF/gH+AgIB
+      /f39/gIF+gH1/f3+Af4CAfoB+f39/gYB/fn5+gIB+fX5+gIJ+e4CCf4F+XYpcjnmGcJBxk4N9fI
+      R7qIiphox+fIWLm52kkoh/fIGXgJd7iH2dbKBkhXaHb4tkfn1+gX1fa1VzcoeFdnhlc3R8bXxQh
+      mOMg395hmySdo97knuTgI6CkYWTiZKKkJCKkYiPh5iBkn+Pfqhvn2t9gYB9jGaHcIBwfGZ/dHNp
+      ZFpzc3N5Y3N3gYOAY39TkGuSgoB8f22ecqp8joKOhp5+hYF8kJOPjZKAlIeQg5d5kX6PeKJnlmy
+      Bc4hshW18a4BrfXBwaXVzhIVxdVJvY311gGSGb4WFfHOMbZd7iXiYfa6Dk354hYqNloyKkIiRiJ
+      J/k4CRfpVyj3mNdplbj2aAd4Njgm19hnlwdWdxd21zbnRufWw=
+    EOS
+
+    result = <<~EOS
+      *5d4d20237a55a6;
+      CRC: 000000\nRSSI: -14.3 dBFS
+      Score: 750\nTime: 206.33us
+      DF:11 AA:4D2023 IID:0 CA:5
+       All Call Reply
+        ICAO Address:  4D2023 (Mode S / ADS-B)
+        Air/Ground:    airborne
+    EOS
+
+    (testpath/"input.bin").write Base64.decode64(enc)
+    assert_equal result.strip, shell_output("#{bin}/readsb --ifile input.bin 2>/dev/null").strip
```

Comment:
Since it's failing, let's see what it has to say in this run.
```suggestion
    assert_equal result.strip, shell_output("#{bin}/readsb --ifile input.bin").strip
```

Reply:
https://github.com/wiedehopf/readsb/issues/94

Most likely a compatibility bug in the program. Its supposed to be backwards compatible with dump1090 which should work with this input.

<img width="798" alt="Screenshot 2025-05-01 at 15 32 09" src="https://github.com/user-attachments/assets/38330458-79d4-4527-8301-fc28270ff747" />

Reply:
No guarantees for full backward compatibility.
It's somewhat compatible with dump1090-mutability not the earlier versions.

And Testing the shell output for this is not a good assert to be honest, i have no intentions of garanteeing that the output is stable.
I suppose you could all lines starting with `*`, those should remain stable.

Reply:
Maybe just something like ``assert_match "ICAO Address:", shell_output("#{bin}/readsb --ifile input.bin")` would be enough.
We just need to know if the command does what we expect it to without errors, we're not looking for a validation of the implementation.

Reply:
Just pick out a couple lines of this:
```
*95318a1d398062807910aef2846d;
```
with sufficient signal strength maybe.
That should be stable.

Reply:
Absolutely, didn't mean to imply any contracts for compatibility. I just explained my understanding of the older program without getting into the weeds.

Thanks for pitching in!

Reply:
I've already answered in the issue opened in readsb, it's currently not working because
```
--device-type ifile
```
is missing as the first option.
Fixed that in the --help now so it's clear.
`readsb --help | grep ifile`

Reply:
I think seeing `ICAO Address:` in the output should be enough of a test to make sure that we're getting some meaningful output.